src/projection_client.rs

```diff
@@ -0,0 +1,629 @@
+                        name: name.as_ref().to_string(),
+                    },
+                )),
+            },
+        )
+        .await
+    }
+
+    pub async fn create_continuous_projection<Name>(
+        &self,
+        name: Name,
+        query: String,
+        track_emitted_streams: bool,
+        credentials: Option<&Credentials>,
+    ) -> crate::Result<()>
+    where
+        Name: AsRef<str>,
+    {
+        self.create_projection(
+            credentials,
+            projections::create_req::Options {
+                query,
+                mode: Some(projections::create_req::options::Mode::Continuous(
+                    projections::create_req::options::Continuous {
+                        name: name.as_ref().to_string(),
+                        track_emitted_streams,
+                    },
+                )),
+            },
+        )
+        .await
+    }
+
+    async fn create_projection(
+        &self,
+        credentials: Option<&Credentials>,
+        options: projections::create_req::Options,
+    ) -> crate::Result<()> {
+        let req = projections::CreateReq {
+            options: Some(options),
+        };
+
+        let mut req = tonic::Request::new(req);
+
+        crate::commands::configure_auth_req(&mut req, credentials.cloned());
+
+        self.client
+            .execute(|handle| async move {
+                let mut client =
+                    projections::projections_client::ProjectionsClient::new(handle.channel);
+                let _ = client.create(req).await?;
+
+                Ok(())
+            })
+            .await
+    }
+
+    pub async fn update_projection<Name>(
+        &self,
+        name: Name,
+        query: String,
+        emit_option: Option<bool>,
+        credentials: Option<&Credentials>,
+    ) -> crate::Result<()>
+    where
+        Name: AsRef<str>,
+    {
+        let options = projections::update_req::Options {
+            name: name.as_ref().to_string(),
+            emit_option: emit_option
+                .map(projections::update_req::options::EmitOption::EmitEnabled)
+                .or(Some(
+                    projections::update_req::options::EmitOption::NoEmitOptions(Empty {}),
+                )),
+            query,
+        };
+
+        let req = projections::UpdateReq {
+            options: Some(options),
+        };
+
+        let mut req = tonic::Request::new(req);
+
+        crate::commands::configure_auth_req(&mut req, credentials.cloned());
+
+        self.client
+            .execute(|handle| async move {
+                let mut client =
+                    projections::projections_client::ProjectionsClient::new(handle.channel.clone());
+
+                let _ = client.update(req).await?;
+
+                Ok(())
+            })
+            .await
+    }
+
+    pub async fn delete_projection<Name>(
+        &self,
+        name: Name,
+        delete_emitted_streams: bool,
```

Comment:
**CONCERN:** In NodeJS client delete params are optional and set the default to `true`. This seems to not be consistent.

Reply:
This is potentially a wrong default from the NodeJS client. Rust doesn't support default value neither.